File: wifi/src/main/java/danielfilho/ufc/br/com/predetect/receivers/NetworkReceiverImpl.java
package danielfilho.ufc.br.com.predetect.receivers;

import android.annotation.SuppressLint;
import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

import br.ufc.quixada.predetect.common.interfaces.NetworkReceiver;
import danielfilho.ufc.br.com.predetect.managers.NetworkManager;

/**
 *
 * @author Daniel Filho
 * @since 2016
 *
 * Updated by Gabriel Cesar, 2018
 *
 */
public class NetworkReceiverImpl extends BroadcastReceiver {

    private NetworkReceiver networkManager;

    public NetworkReceiverImpl() {
        networkManager = NetworkManager.getInstance();
    }

    @Override
    @SuppressLint("UnsafeProtectedBroadcastReceiver")
    public void onReceive(Context context, Intent intent) {
        networkManager.onNetworkReceive(context, intent);
    }

}

File: ble/src/main/kotlin/br/ufc/predetect/ble/receivers/NetworkReceiverImpl.kt
package br.ufc.predetect.ble.receivers

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import br.ufc.predetect.ble.managers.BLENetworkManager
import br.ufc.quixada.predetect.common.interfaces.NetworkReceiver

/**
 * @author Gabriel Cesar
 * @since 2018
 *
 */
class NetworkReceiverImpl(private val networkManager: NetworkReceiver = BLENetworkManager) : BroadcastReceiver() {

    override fun onReceive(context: Context, intent: Intent) = networkManager.onNetworkReceive(context, intent)

}
